src/test/java/com/xebia/headerbuddy/e2e/HeaderBuddyApiKeyTest.java

```diff
@@ -0,0 +1,62 @@
+package com.xebia.headerbuddy.e2e;
+
+import com.xebia.headerbuddy.models.entities.Euser;
+import com.xebia.headerbuddy.models.entities.repositories.EuserRepository;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.web.client.TestRestTemplate;
+import org.springframework.http.ResponseEntity;
+import org.springframework.test.context.junit4.SpringRunner;
+
+import java.util.Date;
+
+// for the annotations
+@RunWith(SpringRunner.class)
+// Run spring on a random port
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+public class HeaderBuddyApiKeyTest {
+
+    @Autowired
+    private TestRestTemplate template;
+    @Autowired
+    private EuserRepository userRepository;
+
+    // Get the random port spring is running on
+    @Value("${local.server.port}")
+    private int port;
+
+    private String key = "abc";
+    private String email = "m@m.nl";
+
+    @Value("${test.http.url}")
+    private String testedUrl;
+
+    // Add the user for the api key
+    @Before
+    public void init(){
+        Euser u = new Euser(key, email);
+        u.setCreationdate(new Date());
+        userRepository.save(u);
+    }
+
+    @Test
+    public void HeaderBuddyCorrectApiKeyTest() {
+        String url = "http://localhost:"+port+"/headerbuddy/api?key="+ key +"&url=" + testedUrl;
```

Comment:
You can use  `String url = "/headerbuddy/api?key="+ key +"&url=" + testedUrl; ` 
Spring Boot wil automatically take care of the host and port